addons/sale/models/account_invoice.py

```diff
@@ -60,7 +60,8 @@ def _onchange_partner_id(self):
     @api.onchange('invoice_user_id')
     def onchange_user_id(self):
         if self.invoice_user_id and self.invoice_user_id.sale_team_id:
-            self.team_id = self.invoice_user_id.sale_team_id
+            st_company = self.sudo().invoice_user_id.sale_team_id.company_id
+            self.team_id = self.invoice_user_id.sale_team_id if not st_company or st_company == self.invoice_user_id.company_id else False
```

Comment:
You should compare with the invoice's company, not the one set on the user.

```suggestion
        if self.sudo().invoice_user_id.sale_team_id.company_id == self.company_id:
            self.team_id = self.invoice_user_id.sale_team_id
```

Having `sale_team_id` on `res.users` looks like a bad implementation. This issue is probably there everywhere so I'm not sure this is something we need to change specificaly in invoices.